replications/internal/queue_management_test.go

```diff
@@ -30,6 +32,81 @@ func TestCreateNewQueueDirExists(t *testing.T) {
 	require.DirExists(t, filepath.Join(queuePath, id1.String()))
 }
 
+func TestEnqueueScanLog(t *testing.T) {
+	t.Parallel()
+
+	// Initialize queue manager with zap observer (to allow assertions on log messages)
+	enginePath, err := os.MkdirTemp("", "engine")
+	require.NoError(t, err)
+	queuePath := filepath.Join(enginePath, "replicationq")
+
+	observedZapCore, observedLogs := observer.New(zap.InfoLevel)
+	observedLogger := zap.New(observedZapCore)
+
+	qm := NewDurableQueueManager(observedLogger, queuePath)
+	defer os.RemoveAll(filepath.Dir(queuePath))
+
+	// Create new queue
+	err = qm.InitializeQueue(id1, maxQueueSizeBytes)
+	require.NoError(t, err)
+
+	// Enqueue some data
+	testData := "weather,location=us-midwest temperature=82 1465839830100400200"
+	err = qm.EnqueueData(id1, []byte(testData))
+	require.NoError(t, err)
+
+	// Give it a second to scan the queue
```

Comment:
Is this wait necessary, since the scanner should be triggered to run right away when data is added to the queue?

Reply:
I'm wondering if instead of waiting for a time, could we instead we call `qm.Close()` and wait for that to return. Since that would be called after the `EnqueueData`, I believe it would not return until `rq.run` had finished processing that receive.

Reply:
Hmm that's true - it should be scanning right away. It seems like that isn't happening now, because without the wait, no log messages are observed. But when I add the 1 second sleep, the log messages are present (data is being scanned before we make assertions on the logs). @DStrand1 we probably need to remove the sleep and take a look at why that 1 second delay is happening.

Reply:
Your second comment wasn't showing up for me at first - got it, that seems like a better way to do it.